src/sentry/db/mixin.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import
+
+import logging
+import six
+
+from uuid import uuid4
+
+from sentry.models import OrganizationOption
+
+logger = logging.getLogger('sentry.deletions')
+
+
+def delete_pending_deletion_option(instance, **kwargs):
+    if hasattr(instance, 'delete_pending_deletion_option'):
+        instance.delete_pending_deletion_option()
+
+
+class PendingDeletionMixin(object):
+    _rename_fields_on_pending_delete = frozenset()
+
+    def build_pending_deletion_key(self):
+        return 'pending-delete:%s:%s' % (self.__class__.__name__, self.id)
+
+    def rename_on_pending_deletion(self, fields=None):
+        fields = fields or self._rename_fields_on_pending_delete
+        original_data = {}
+
+        for field in fields:
+            original_data[field] = getattr(self, field)
+            setattr(self, field, uuid4().hex)
+
+        self.save()
+        original_data['id'] = self.id
+        original_data['model'] = self.__class__.__name__
+        OrganizationOption.objects.create(
+            organization_id=self.organization_id,
+            key=self.build_pending_deletion_key(),
+            value=original_data,
+        )
```

Comment:
Do you need this OrganizationOption now? The renamed fields are stored in the `repository.config` metadata.